### PR TITLE
ListRepo() gRPC pagination [CORE-2132]

### PR DESCRIPTION
**Pull request by acohen4**

### CODE REVIEW

File: src/pfs/pfs.proto [hunk 1]
```diff
@@ -200,6 +200,17 @@ message ListRepoRequest {
   string type = 1;
   // projects filters out repos that do not belong in the list, while no projects means list all repos.
   repeated Project projects = 2;
+  // repo page defines the
```
Comment (acohen4) [inline src/pfs/pfs.proto:203]: need to fix this comment

File: src/internal/pfsdb/common.go [hunk 2]
```diff
@@ -47,26 +47,33 @@ func OrderByQuery[T ColumnName](orderBys ...OrderByColumn[T]) string {
...
+	limit, offset, maxPages uint64
+	page                    []T
+	pageIdx                 int
+	lastTimestamp           time.Time
+	revision                int64
+	pagesSeen               int
 }
 
-func newPageIterator[T ModelType](ctx context.Context, query string, values []any, startPage, pageSize uint64) pageIterator[T] {
+// if pages == 0, then interpret as unlimited pages
```
Comment (FahadBSyed) [inline src/internal/pfsdb/common.go:60]: This should be `maxPages` right?

**Review by kevyang — APPROVED** [review]
rubber stamp for integrations

**Review by FahadBSyed — APPROVED** [review]
I was thinking about the page parameter in ForEachRepo() and in hindsight I wish we had grouped the parameters better during the postgres migration since the signature is starting to get complicated. 

That is out of scope for your change though. I think given the constraints of the codebase,  this is probably the best way to implement paging.

**Pull request merged**